pkg/runtime/runtime.go

```diff
@@ -305,6 +315,40 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 	return nil
 }
 
+func (a *DaprRuntime) populateSecretsConfiguration() {
+	for name := range a.secretStores {
+		// This is added for the scenario where secret store component is added, but
+		// no restrictive configuration is defined.
+		a.defaultSecretAccess[name] = config.AllowAccess
+		for _, storeInList := range a.globalConfig.Spec.Secrets.Scopes {
+			if storeInList.StoreName == name {
+				// Accept only "allow" or "deny" strings or default to "allow".
+				switch strings.ToLower(storeInList.DefaultAccess) {
+				case config.AllowAccess:
+					a.defaultSecretAccess[name] = config.AllowAccess
+				case config.DenyAccess:
+					a.defaultSecretAccess[name] = config.DenyAccess
+				default:
+					a.defaultSecretAccess[name] = config.AllowAccess
+				}
+				// if deny, break. Eg: kubernetes is put as deny access with no allowed list.
+				// This scenario also ignores only deniedSecrets being populated if the defaultAccess is deny.
+				if a.defaultSecretAccess[name] == config.DenyAccess && storeInList.AllowedSecrets == nil {
+					break
+				}
+				if storeInList.AllowedSecrets != nil {
+					//Convert to a Set like access for easy lookup.
+					a.allowedSecrets[name] = convertToMap(storeInList.AllowedSecrets)
```

Comment:
May be try https://godoc.org/k8s.io/apimachinery/pkg/util/sets#NewString which has these utilities/helpers already? `sets.String` may also be more readable than `map[string]interface{}` or `map[string]struct{}`